src/mbgl/renderer/layers/render_location_indicator_layer.cpp

```diff
@@ -539,10 +546,12 @@ void main() {
                 params.perspectiveCompensation; // Compensation factor for the perspective deformation
         //     ^ clamping this to 0.8 to avoid growing the puck too much close to the camera.
         const double shadowRadius =
-            params.puckShadowSizePx * M_SQRT2 * 0.5 *
+            ((texShadow) ? texShadow->width : 0.0) * params.puckShadowScale * M_SQRT2 * 0.5 *
```

Comment:
Could you test with '@2x' and '@3x' images? We need to scale quads based on image's pixel ratio.

Reply:
added support for dividing the width by the image pixel ratio